lua/wikis/commons/ThisDay.lua

```diff
@@ -0,0 +1,347 @@
+		:add{
+			ConditionNode(ColumnName('birthdate_month'), Comparator.eq, month),
+			ConditionNode(ColumnName('birthdate_day'), Comparator.eq, day),
+			ConditionNode(ColumnName('deathdate'), Comparator.eq, DateExt.defaultDate),
+			ConditionNode(ColumnName('birthdate'), Comparator.neq, DateExt.defaultDate),
+		}
+
+	return mw.ext.LiquipediaDB.lpdb('player', {
+		limit = 5000,
+		conditions = conditions:toString(),
+		query = 'extradata, pagename, id, birthdate, nationality, links',
+		order = 'birthdate asc, id asc'
+	})
+end
+
+--- Queries patch data
+---@param month integer
+---@param day integer
+---@return datapoint[]
+function Query.patch(month, day)
+	local conditions = ConditionTree(BooleanOperator.all)
+		:add{
+			ConditionNode(ColumnName('date'), Comparator.neq, DateExt.defaultDate),
+			ConditionNode(ColumnName('date_month'), Comparator.eq, month),
+			ConditionNode(ColumnName('date_day'), Comparator.eq, day),
+			ConditionNode(ColumnName('type'), Comparator.eq, 'patch'),
+		}
+
+	return mw.ext.LiquipediaDB.lpdb('datapoint', {
+		limit = 5000,
+		conditions = conditions:toString(),
+		query = 'pagename, name, date',
+		order = 'date asc, name asc'
+	})
+end
+
+--- Queries tournament win data
+---@param month integer
+---@param day integer
+---@return placement[]
+function Query.tournament(month, day)
+	local conditions = ConditionTree(BooleanOperator.all)
+		:add{
+			ConditionNode(ColumnName('date'), Comparator.neq, DateExt.defaultDate),
+			ConditionNode(ColumnName('date_month'), Comparator.eq, month),
+			ConditionNode(ColumnName('date_day'), Comparator.eq, day),
+			ConditionNode(ColumnName('date'), Comparator.lt, os.date('%Y-%m-%d', os.time() - 86400)),
+			ConditionNode(ColumnName('placement'), Comparator.eq, 1),
+			ConditionNode(ColumnName('opponentname'), Comparator.neq, 'TBD'),
+			ConditionTree(BooleanOperator.any)
+				:add{ConditionNode(ColumnName('prizepoolindex'), Comparator.eq, '1')}
+		}
+	conditions:add(Query._multiValueCondition(
+		'liquipediatier',
+		Config.tiers,
+		BooleanOperator.any
+	))
+	conditions:add(Query._multiValueCondition(
+		'liquipediatiertype',
+		Config.tierTypes,
+		BooleanOperator.all
+	))
+
+	return mw.ext.LiquipediaDB.lpdb('placement', {
+		limit = 5000,
+		conditions = conditions:toString(),
+		query = 'extradata, pagename, date, icon, icondark, shortname, tournament, series, '
+			.. 'opponentname, opponenttemplate, opponentplayers, opponenttype',
+		order = 'date asc, pagename asc'
+	})
+end
+
+--- build conditions for multi variable
+---@param key string
+---@param values table
+---@param booleanOperator lpdbBooleanOperator
+---@return table?
+function Query._multiValueCondition(key, values, booleanOperator)
+	if Table.isEmpty(values) then
+		return
+	end
+
+	local conditions = ConditionTree(booleanOperator)
+
+	for _, value in pairs(values) do
+		conditions:add{ConditionNode(ColumnName(key), Comparator.eq, value)}
+	end
+
+	return conditions
+end
```

Comment:
do we deem this usefull for other modules?
if so could possibly move it to Module:Condition